src/Analyzers/CSharp/Analyzers/UseCollectionExpression/CSharpUseCollectionExpressionForFluentDiagnosticAnalyzer.cs

```diff
@@ -59,6 +58,15 @@ internal sealed partial class CSharpUseCollectionExpressionForFluentDiagnosticAn
         nameof(ImmutableStack<int>),
         nameof(System.Collections.Immutable));
 
+    /// <summary>
+    /// Set of type-names that are blocked from moving over to collection expressions because the semantics of them are
+    /// known to be specialized, and thus could change semantics in undesirable ways if the compiler emitted its own
+    /// code as an replacement.
+    /// </summary>
+    private static readonly ImmutableHashSet<string?> s_bannedTypes = ImmutableHashSet.Create<string?>(
+        nameof(ParallelEnumerable),
+        nameof(ParallelQuery));
```

Comment:
It felt weird that we aren't doing namespace checks here, but I guess if anybody else calls their types these names, we probably should stay away too.

Reply:
yup.